packages/block-library/src/separator/edit.js

```diff
@@ -6,20 +6,28 @@ import clsx from 'clsx';
 /**
  * WordPress dependencies
  */
-import { HorizontalRule } from '@wordpress/components';
+import { HorizontalRule, SelectControl } from '@wordpress/components';
 import {
 	useBlockProps,
 	getColorClassName,
 	__experimentalUseColorProps as useColorProps,
+	InspectorControls,
 } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import useDeprecatedOpacity from './use-deprecated-opacity';
 
+const htmlElementMessages = {
+	div: __(
+		'The <div> element should only be used if the separator is a design element that should not be announced.'
+	),
+};
+
 export default function SeparatorEdit( { attributes, setAttributes } ) {
-	const { backgroundColor, opacity, style } = attributes;
+	const { backgroundColor, opacity, style, tagName } = attributes;
```

Comment:
```suggestion
	const { backgroundColor, opacity, style, tagName = 'hr' } = attributes;
```

Assigning a default value here should fix unit tests.

Reply:
I'm not sure if we should assign a default here, it's assigned by the block parsing instead. I guess we have some old deprecation migrations where the tagName is not set properly. I'll see if there's another way.

Reply:
I think you can also update the `defaultAttributes` object in unit test to mimic actual behavior.

Reply:
I think it's solved.

Reply:
IMO, there should be no need for deprecation or to update it. The enhancement doesn't change the default behavior.

Reply:
I disagree, we should update the deprecation if the shape of the expected "attributes" is changed, and it's changed now since the attribute is mandatory.

Reply:
The `save` function is unchanged, so we don't need a new deprecation but for old deprecation that have "migrate" functions we need to update them to the correct attribute shape.

Reply:
We have not handled it that way in the past; if attribute shape changes, a new deprecation is created. Maybe it only worked because default attributes don't get serialized in content.

@ntsekouras, what do you think?

Reply:
Same as the other comment. Current shape looks good.